server/src/main/java/org/elasticsearch/search/vectors/KnnScoreDocQuery.java

```diff
@@ -83,7 +88,11 @@ public Explanation explain(LeafReaderContext context, int doc) {
 
             @Override
             public Scorer scorer(LeafReaderContext context) {
-
+                // Segment starts indicate how many docs are in the segment,
+                //   upper equalling lower indicates no documents for this segment
+                if (segmentStarts[context.ord] == segmentStarts[context.ord + 1]) {
+                    return null;
+                }
```

Comment:
Again, if upper and lower segment starts are equal, there are no docs in this particular segment. So just return null.